lib/whatsapp_sdk/resource/interactive_action_button.rb

```diff
@@ -0,0 +1,126 @@
+# typed: strict
+# frozen_string_literal: true
+
+module WhatsappSdk
+  module Resource
+    class InteractiveActionButton
+      extend T::Sig
+
+      class InvalidType < StandardError
+        extend T::Sig
+
+        sig { returns(String) }
+        attr_reader :message
+
+        sig { params(type: String).void }
+        def initialize(type)
+          @message = T.let(
+            "invalid type #{type} for interactive. type should be reply only.",
+            String
+          )
+          super(message)
+        end
+      end
```

Comment:
How do you feel if we extract this class into a generic error in its own file like this? 

```ruby
class ErrorWithMessage < StandardError
  extend T::Sig

  sig { returns(String) }
  attr_reader :message

  sig { params(message: String).void }
  def initialize(message)
    @message = T.let(
      "invalid type #{type} for interactive. type should be reply only.",
      String
    )
    super(message)
  end
end
```

Then in each file, we can reuse it.

Reply:
I'm ok with doing it in a separate PR if it's getting too big.

Reply:
Help me understand the purpose of `ErrorWithMessage`, as I see it is simply `StandardError` as it doesn't have anything other than `message`. Should we DRY out specifically `InvalidType` error from multiple definitions?

Reply:
Yeah, the idea is we can reuse it since it's basically the same across all classes but just a different message.

Reply:
Can we use something like
```ruby
module WhatsappSdk
  module Resource
    module Error
      class InvalidType < WhatsappSdk::Error
        extend T::Sig

        sig { returns(String) }
        attr_reader :message

        sig { params(type: String, klass: String, valid_types: Array).void }
        def initialize(type, klass, valid_types)
          @message = T.let(
            "invalid type #{type} for #{klass}. type should be #{valid_types.map(&:serialize).join(', ')} only.",
            String
          )
          super(message)
        end
      end
    end
  end
end
```

Although the more I'm trying it out, `InvalidType` seems to be an error we should never need as sorbet `T::Enum` should take care of the type and incorrect value. Should I go ahead and remove it?

Reply:
I've taken a stab at it, please check if the move was in the right direction.